### ISSUE TITLE
Ctrl-Tab and select all occurrences in file

### ISSUE THREAD

**Issue by fdncred** [ProgerXP/Notepad2e#255]
I have `HighlightSelection=1` and `MaxSearchDistance=10000`. When I hit Ctrl-Tab I get items selected but only the items on my screen. I figured upping MaxSearchDistance to 10k would help this but it doesn't appear to. 

How can I make Ctrl-Tab highlight/make editable all occurrences of the selection in the entire file vs just what is on screen?

Thanks,
Darren

**Comment by ProgerXP** [comment]
MaxSearchDistance only affects how far occurrences are checked when deciding between highlighting styles (Page, Single). It doesn't affect Ctrl+Tab - that one always edits visible occurrences on the page.

> How can I make Ctrl-Tab highlight/make editable all occurrences of the selection in the entire file vs just what is on screen?

We have no such feature. Why don't you use Find & Replace for this (Ctrl+H)? It has about the same number of keystrokes:

- select what you want to replace
- Ctrl+H 
- Tab
- enter replacement 
- Alt+A 

If you check `Close after replace` and `Don't display this message again` then you don't have to press Escape and Tab becomes the only extra keystroke.

**Comment by fdncred** [comment]
Why even have the Ctrl-Tab if you can't use it for the whole file? Of course, one could use Ctrl-H. That wasn't the point. Maybe I'll pull the code and see if it's an easy fix. Thanks!

**Comment by ProgerXP** [comment]
> Why even have the Ctrl-Tab if you can't use it for the whole file? 

In contrast, what would be the use of duplicating the same Find & Replace functionality? Ctrl+Tab is used to rename variables and such. It's generally a bad idea to rename something that you do not see due to unpredictable results. If you want to replace just everything then you don't require highlighting throughout the entire buffer (since you won't see it anyway) and so Ctrl+H is your friend - and it will work much faster too.

Basically Ctrl+Tab guarantees that it will not alter contents that you do not see so it's safer (there's a catch though - it still counts long wrapped sublines that are invisible).

**Comment by fdncred** [comment]
For me, I like the ability to have multiple cursors and watch the changes being made as I type, granted I can only see one screen full at a time. But I don't want to have to do it again on every screen full to make the changes. You are right that I could just use the Ctrl-H, and I do that sometimes. But if you're going to have a Ctrl-Tab functionality at lease give me that ability to do it on the entire file, via an ini file change, if I want to take that risk. It's such a cool feature, I just want to use it. Thanks.

**Comment by ProgerXP** [comment]
@cshnik Does current Edit Word mechanism painlessly support removing "only visible" restriction via a setting? Also consider other related issues (#246 #248).

**Comment by cup** [comment]
https://github.com/ProgerXP/Notepad2e/issues/211
https://github.com/ProgerXP/Notepad2e/issues/246

https://github.com/ProgerXP/Notepad2e/issues/255
